Project/Source/Modules/ModuleUserInterface.cpp

```diff
@@ -0,0 +1,206 @@
+	//Right now we have FileDialog, it may change in the future
+	//FileDialog::GetFileName(fontPath.c_str());
+
+	//OLD VERSION, TO DO USE NEW SYSTEM, AS STATED IN THE PREVIOUS COMMENT
+	/*
+	std::string fontName = FileDialog::GetFileName(fontPath.c_str());
+
+	std::unordered_map<std::string, ResourceFont*>::const_iterator existsKey = fonts.find(fontName);
+
+	if (existsKey == fonts.end()) {
+		ResourceFont* font = FontImporter::ImportFont(fontPath);
+		//FontImporter::LoadFont(fontPath, characters);
+		if (font) { // TODO: This is a bad check right now. It will always be initialized. Change it when we have the resource manager
+			font->name = fontName;
+			fonts.insert(std::pair<std::string, ResourceFont*>(fontName, font));
+		} else {
+			LOG("Couldn't load font %s", fontPath.c_str());
+		}
+	}
+	*/
+}
+
+Character ModuleUserInterface::GetCharacter(UID font, char c) { // Should this return a Character*?
+	//std::unordered_map<std::string, UID>::const_iterator existsKey = fonts.find(font);
+	ResourceFont* fontResource = (ResourceFont*) App->resources->GetResource(font);
+
+	if (fontResource == nullptr) {
+		return Character();
+	}
+	return fontResource->characters[c];
+}
+
+void ModuleUserInterface::GetCharactersInString(UID font, const std::string& sentence, std::vector<Character>& charsInSentence) {
+	ResourceFont* fontResource = (ResourceFont*) App->resources->GetResource(font);
+
+	if (fontResource == nullptr) {
+		return;
+	}
+
+	for (std::string::const_iterator i = sentence.begin(); i != sentence.end(); ++i) {
+		charsInSentence.push_back(fontResource->characters[*i]);
+	}
+}
+
+void ModuleUserInterface::Render() {
+	Scene* scene = App->scene->scene;
+	if (scene != nullptr) {
+		for (ComponentCanvasRenderer canvasRenderer : scene->canvasRendererComponents) {
+			canvasRenderer.Render(&canvasRenderer.GetOwner());
+		}
+	}
+}
+
+GameObject* ModuleUserInterface::GetCanvas() const {
+	return canvas;
+}
+
+void ModuleUserInterface::ReceiveEvent(const Event& e) {
+	float2 mousePos = float2(e.mouseUpdate.mouseX, e.mouseUpdate.mouseY);
+
+	switch (e.type) {
+	case EventType::MOUSE_UPDATE:
+		if (currentEvSys) {
+			for (ComponentSelectable& selectable : App->scene->scene->selectableComponents) {
+				ComponentBoundingBox2D* bb = selectable.GetOwner().GetComponent<ComponentBoundingBox2D>();
+
+				if (!selectable.IsHovered()) {
+					if (bb->GetWorldAABB().Contains(mousePos)) {
+						selectable.OnPointerEnter();
+					}
+				} else {
+					if (!bb->GetWorldAABB().Contains(mousePos)) {
+						selectable.OnPointerExit();
+					}
+				}
+			}
+		}
+		break;
+
+	case EventType::MOUSE_CLICKED:
+		if (currentEvSys != nullptr) {
+			ComponentSelectable* lastHoveredSelectable = currentEvSys->GetCurrentlyHovered();
+			if (lastHoveredSelectable != nullptr) {
+				if (lastHoveredSelectable->IsInteractable()) {
+					IMouseClickHandler* i = dynamic_cast<IMouseClickHandler*>(lastHoveredSelectable->GetSelectableComponent());
```

Comment:
- i -> mouseClickHandler
- dynamic_cast está totalmente prohibido. Es demasiado lento. En este caso podéis hacer un cast normal o static_cast.